testConnectivity.py

```diff
@@ -243,14 +245,59 @@ def testConnectivity(args):
         vfatBoard.parentOH.parentAMC.writeRegister("GEM_AMC.TTC.GENERATOR.ENABLE",0x0)
 
         print("Checking trigger link status:")
-        for trial in range(0,args.maxIter ):
-            testLinks = vfatBoard.parentOH.parentAMC.getTriggerLinkStatus(printSummary=True, checkCSCTrigLink=args.checkCSCTrigLink, ohMask=args.ohMask)
-	    if (testLinks < 1):
-                printGreen("Trigger link to OHs in mask:{0} is good".format(hex(args.ohMask)))
+        for trial in range(0,args.maxIter):
+            testLinks = vfatBoard.parentOH.parentAMC.getTriggerLinkStatus(
+                            printSummary=True, 
+                            checkCSCTrigLink=args.checkCSCTrigLink, 
+                            ohMask=args.ohMask)
+            isDead = True
+            listOfDeadFPGAs = []
+            for ohN in range(nOHs):
+                # Skip masked OH's
+                if( not ((args.ohMask >> ohN) & 0x1)):
+                    continue
+
+                # Check Trigger Link Status
+                if args.checkCSCTrigLink:
+                    if (testLinks[ohN] == 0 and testLinks[ohN+1] == 0): # All Good
```

Comment:
testLinks with the argument checkCSCTrigLink=True will return for (ohN && ohN+1). Having here 'and testLinks[ohN+1]' along with args.checkCSCTrigLink=True would therefore check (ohN && ohN+1) and (ohN+1 && ohN+2)

Reply:
The loop is only looping over unmasked OH's both here and in the `HwAMC` method.  And this is not incorrectly considering the next `OH`, see for example this call:

```bash
% testConnectivity.py -d --shelf=2 -s5 -o 0x10 --nPhaseScans=100 --skipDACScan --skipScurve --checkCSCTrigLink -a
...
...
====================
Step 3: Programming FPGA & Checking FPGA Communication
====================
Initial value to write: 1, register GEM_AMC.TTC.GENERATOR.ENABLE
Initial value to write: 1, register GEM_AMC.TTC.GENERATOR.SINGLE_HARD_RESET
Initial value to write: 0, register GEM_AMC.TTC.GENERATOR.ENABLE
Checking trigger link status:
--=======================================--
-> GEM SYSTEM TRIGGER LINK INFORMATION
--=======================================--

----------OH4----------
LINK0_MISSED_COMMA_CNT          0xffff
LINK1_MISSED_COMMA_CNT          0x0
LINK0_OVERFLOW_CNT              0x0
LINK1_OVERFLOW_CNT              0x0
LINK0_UNDERFLOW_CNT             0xffff
LINK1_UNDERFLOW_CNT             0x5
LINK0_SBIT_OVERFLOW_CNT         0xffff
LINK1_SBIT_OVERFLOW_CNT         0x0
----------OH5----------
LINK0_MISSED_COMMA_CNT          0x0
LINK1_MISSED_COMMA_CNT          0xffff
LINK0_OVERFLOW_CNT              0x0
LINK1_OVERFLOW_CNT              0x0
LINK0_UNDERFLOW_CNT             0x4
LINK1_UNDERFLOW_CNT             0xffff
LINK0_SBIT_OVERFLOW_CNT         0x0
LINK1_SBIT_OVERFLOW_CNT         0xffff
Trigger link of OHs: [4, 5] failed, but I was told to accept bad trigger links
FPGA Communication Established
```

Here with `ohMask = 0x10` and `--checkCSCTrigLink` it's only querying OH4 and OH5 as expected, it does not then later query OH5 & OH6.

Reply:
### ok i think this is fixed when you then changed xhal [here](https://github.com/cms-gem-daq-project/cmsgemos/pull/269/files#diff-b4c55813312aea84fc2ab6fbe9b99365R436) right?